Source/WebKit/UIProcess/Gamepad/gtk/UIGamepadProviderGtk.cpp

```diff
@@ -71,6 +71,14 @@ static WebPageProxy* getWebPageProxy(GtkWidget* widget)
 WebPageProxy* UIGamepadProvider::platformWebPageProxyForGamepadInput()
 {
     GUniquePtr<GList> toplevels(gtk_window_list_toplevels());
+
+    // Work around for WebKitTestRunner.
+    // If there's only one toplevel window available, try to get WebPageProxy from that window.
+    if (g_list_length(toplevels.get()) == 1) {
+        WebPageProxy* proxy = getWebPageProxy(GTK_WIDGET(toplevels.get()->data));
+        return proxy ? proxy : nullptr;
+    }
```

Comment:
I don't like this workaround. IIRC the issue was related to how `WebCore::widgetIsOnscreenToplevelWindow()` is not working as expected in GTK4. I think we need to fix that instead.

Reply:
No, the problem is that xvfb doesn't support toplevel focus. We handle this in other cases, so here we should do the same. I'll submit a patch.

Reply:
See https://github.com/WebKit/WebKit/pull/47710